src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> newNode;
+    private Node<T> last;
+    private Node<T> first;
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.element = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
-    public void add(T value) {
+    public void add(T element) {
+        linkLast(element);
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T element, int index) {
+        if (index == size) {
+            linkLast(element);
+        } else {
+            checkIndexPos(index);
+            linkBefore(element, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T current : list) {
+            linkLast(current);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexPos(index);
+        return findNodeByIndex(index).element;
     }
 
     @Override
-    public T set(T value, int index) {
-        return null;
+    public T set(T element, int index) {
+        checkIndexPos(index);
+        Node<T> x = findNodeByIndex(index);
+        T original = x.element;
+        x.element = element;
+        return original;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexPos(index);
+        return unlink(findNodeByIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (node.element == null) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (object.equals(node.element)) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        }
```

Comment:
we don't need "if-else" here and we can have one "for loop" if we change "if" statement inside the loop like this:

if (object == node || (object != null && object.equals(node.element)))